src/Corax/IndexWriter.cs

```diff
@@ -29,14 +30,106 @@ public enum TermIdMask : long
         Set = 2
     }
 
+
+    public class IndexFieldsMapping : IEnumerable<IndexFieldBinding>
+    {
+        public static readonly IndexFieldsMapping Instance = new IndexFieldsMapping(null);
+
+        private readonly ByteStringContext _context;
+        private readonly Dictionary<Slice, IndexFieldBinding> _fields;
+        private readonly Dictionary<int, IndexFieldBinding> _fieldsById;
```

Comment:
What's the life time of those? Shouldn't we be cloning the `Slice` ?

Reply:
The lifetime of this is tied to the actual context. They are created on line #45. `IndexFieldsMapping` does not decide itself what the lifetime is, this is the responsibility of the caller. I foresee though that this mappings will be unique to the Index itself, so they will be created only once.

Reply:
That means that the caller should ensure that the `IndexFieldsMapping`'s context is as long lived, no?
Just something to make sure that we do

Reply:
The best scenario would be a single instance per index and processor (with affinity).